From d2f338d85c720c397442b37e6d4f68b93e9e9eec Mon Sep 17 00:00:00 2001
From: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
Date: Tue, 21 Apr 2026 14:53:04 -0300
Subject: [PATCH] ceph-csi-rbd: fix snapshotclass names

In cephcsi 3.15.0 the snapshotclass.yaml file already exists
but a few field names are different than what we used.

We need to fix those names for backwards compatibility
on upgrade and rollback.

Signed-off-by: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
---
 .../ceph-csi-rbd/templates/snapshotclass.yaml | 25 ++++++++++---------
 charts/ceph-csi-rbd/values.yaml               |  6 ++---
 2 files changed, 16 insertions(+), 15 deletions(-)

diff --git a/charts/ceph-csi-rbd/templates/snapshotclass.yaml b/charts/ceph-csi-rbd/templates/snapshotclass.yaml
index d496816..dcdf445 100644
--- a/charts/ceph-csi-rbd/templates/snapshotclass.yaml
+++ b/charts/ceph-csi-rbd/templates/snapshotclass.yaml
@@ -1,11 +1,11 @@
-{{- if .Values.volumeSnapshotClass.create -}}
+{{- if .Values.snapshotClass.create -}}
 apiVersion: snapshot.storage.k8s.io/v1
 kind: VolumeSnapshotClass
 metadata:
-  name: {{ .Values.volumeSnapshotClass.name }}
-{{- if .Values.volumeSnapshotClass.annotations }}
+  name: {{ .Values.snapshotClass.name }}
+{{- if .Values.snapshotClass.annotations }}
   annotations:
-{{ toYaml .Values.volumeSnapshotClass.annotations | indent 4 }}
+{{ toYaml .Values.snapshotClass.annotations | indent 4 }}
 {{- end }}
   labels:
     app: {{ include "ceph-csi-rbd.name" . }}
@@ -13,18 +13,19 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
     {{- with .Values.commonLabels }}{{ toYaml . | trim | nindent 4 }}{{- end }}
-    {{- with .Values.volumeSnapshotClass.labels }}{{ toYaml . | trim | nindent 4 }}{{- end }}
+    {{- with .Values.snapshotClass.labels }}{{ toYaml . | trim | nindent 4 }}{{- end }}
 driver: {{ .Values.driverName }}
 parameters:
-  clusterID: {{ .Values.volumeSnapshotClass.clusterID }}
-{{- if .Values.volumeSnapshotClass.snapshotNamePrefix }}
-  snapshotNamePrefix: "{{ .Values.volumeSnapshotClass.snapshotNamePrefix }}"
+  clusterID: {{ .Values.snapshotClass.clusterID }}
+{{- if .Values.snapshotClass.snapshotNamePrefix }}
+  snapshotNamePrefix: "{{ .Values.snapshotClass.snapshotNamePrefix }}"
 {{- end }}
-  csi.storage.k8s.io/snapshotter-secret-name: {{ .Values.volumeSnapshotClass.snapshotterSecret }}
-{{- if .Values.volumeSnapshotClass.snapshotterSecretNamespace }}
-  csi.storage.k8s.io/snapshotter-secret-namespace: {{ .Values.volumeSnapshotClass.snapshotterSecretNamespace }}
+  csi.storage.k8s.io/snapshotter-secret-name: {{ .Values.snapshotClass.provisionerSecret }}
+{{- if .Values.snapshotClass.provisionerSecretNamespace }}
+  csi.storage.k8s.io/snapshotter-secret-namespace: {{ .Values.snapshotClass.provisionerSecretNamespace }}
 {{ else }}
   csi.storage.k8s.io/snapshotter-secret-namespace: {{ .Release.Namespace }}
 {{- end }}
-deletionPolicy: {{ .Values.volumeSnapshotClass.deletionPolicy }}
+deletionPolicy: {{ .Values.snapshotClass.deletionPolicy }}
 {{- end -}}
+
diff --git a/charts/ceph-csi-rbd/values.yaml b/charts/ceph-csi-rbd/values.yaml
index 84b8ad9..8154bc6 100644
--- a/charts/ceph-csi-rbd/values.yaml
+++ b/charts/ceph-csi-rbd/values.yaml
@@ -547,7 +547,7 @@ storageClass:
   # Pool chunk size / PG_NUM
   chunk_size: 8

-volumeSnapshotClass:
+snapshotClass:
   # Specifies whether the VolumeSnapshotClass should be created
   create: false
   name: csi-rbdplugin-snapclass
@@ -579,10 +579,10 @@ volumeSnapshotClass:
   snapshotNamePrefix: ""

   # The secrets have to contain user and/or Ceph admin credentials.
-  snapshotterSecret: csi-rbd-secret
+  provisionerSecret: csi-rbd-secret
   # If the Namespaces are not specified, the secrets are assumed to
   # be in the Release namespace.
-  snapshotterSecretNamespace: ""
+  provisionerSecretNamespace: ""

   deletionPolicy: Delete

--
2.34.1
